packages/flutter/lib/src/widgets/sliver_floating_header.dart

```diff
@@ -265,7 +301,10 @@ class _RenderSliverFloatingHeader extends RenderSliverSingleBoxAdapter {
 
     child?.layout(constraints.asBoxConstraints(), parentUsesSize: true);
     final double paintExtent = childExtent - effectiveScrollOffset;
-    final double layoutExtent = childExtent - constraints.scrollOffset;
+    final double layoutExtent = switch(snapMode ?? FloatingHeaderSnapMode.overlay) {
```

Comment:
Linux analyzer found this😄
```suggestion
    final double layoutExtent = switch (snapMode ?? FloatingHeaderSnapMode.overlay) {
```

Reply:
Thanks, fixed.